model.js

```diff
@@ -8,13 +9,54 @@ exports.selectTopics = () => {
 }
 
 exports.selectArticleById = (id) =>{
-    return db.query(`SELECT * FROM articles WHERE article_id = ${id}`).then(({rows}) =>{
+    return db.query(`SELECT * FROM articles WHERE article_id = $1`,[id]).then(({rows}) =>{
         if (rows.length === 0){
             return Promise.reject({
                 status: 404,
-                msg: "Couldn't find article"
+                msg: "Not found"
             })
         }
         return rows[0]
     })
-}
\ No newline at end of file
+}
+
+exports.selectArticles = () =>{
+   const getArticles = () =>{ return db.query(`SELECT * FROM articles ORDER BY created_at DESC`).then(({rows}) =>{
+
+        
+        rows.forEach((article) =>{
+            article.comment_count = 0
+
+            delete article.body
+
+        })
+        return rows
+    })
+}
+
+    const getComments = () =>{ return db.query('SELECT * FROM comments').then(({rows}) => { 
+        
+        
+        return rows
+    })
+    }
+   return Promise.all([getArticles(),getComments()]).then((articlesAndComments)=>{
+
+        articlesAndComments[1].forEach((comment)=>{
+            articlesAndComments[0].forEach((article)=>{
+                let commentCount = 0 
+                if(comment.article_id === article.article_id){
+                    article.comment_count += 1
+                }
+            })
+        })
+        return(articlesAndComments[0])
+    })
```

Comment:
Continuing from my comments above, I can see what you've done to alter the comments but there is no need to make two calls to the database at this stage, plus the additional logic makes this quite inefficient. You will need to make more calls to the comments table later on so this function will be used at that point. Again, message me for any clarification if needed.